### ISSUE TITLE
Google Cloud CPU Platform issues.

### ISSUE THREAD

**Issue by upodroid** [poseidon/typhoon#438]
### Environment

* Platform: google-cloud
* OS:  N/A
* Release: latest?
* Terraform and plugins
```
Terraform v0.11.13
+ provider.ct v0.3.1
+ provider.google v2.2.0
+ provider.local v1.2.0
+ provider.null v1.0.0
+ provider.template v1.0.0
+ provider.tls v1.2.0
```
### Problem

Terraform complains about cpu platform not being available in a particlar region. Intel Haswell patform is not avalable in any of the newer regions such as europe-west2/6 . Haswell is available in europe-west1 which is OK but is there any reason why Container Linux can't run on newer platforms?

Terraform output
```
Error: Error applying plan:

3 error(s) occurred:

* module.google-cloud-yavin.google_compute_instance.controllers[2]: 1 error(s) occurred:

* google_compute_instance.controllers.2: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.minCpuPlatform': 'Intel Haswell'. CPU platform unavailable in requested zone. Valid options for this zone are: Intel Skylake, Intel Broadwell., invalid
* module.google-cloud-yavin.google_compute_instance.controllers[1]: 1 error(s) occurred:

* google_compute_instance.controllers.1: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.minCpuPlatform': 'Intel Haswell'. CPU platform unavailable in requested zone. Valid options for this zone are: Intel Skylake, Intel Broadwell., invalid
* module.google-cloud-yavin.google_compute_instance.controllers[0]: 1 error(s) occurred:

* google_compute_instance.controllers.0: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.minCpuPlatform': 'Intel Haswell'. CPU platform unavailable in requested zone. Valid options for this zone are: Intel Skylake, Intel Broadwell., invalid

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by dghubble** [comment]
Typhoon GCP defines the *minimum* CPU platform as Haswell via [#405](https://github.com/poseidon/typhoon/pull/405). Meaning avoid Sandy/Ivy bridge, but allow Haswell or newer like Broadwell, Skylake, etc.

This sounds like a bug in [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) and logic for comparing newer/older CPU platforms.

**Comment by dghubble** [comment]
Ugh. No issue in terraform-provider-google or compute/v1 [client](https://github.com/googleapis/google-api-go-client/blob/master/compute/v1/compute-gen.go). I think the backend Google Cloud API actually implements `SetMinCpuPlatform` to mean "use exactly that version or error". Rather misleading going by Terraform or API comments.

> Compute Engine always uses the minimum CPU platform where available.

I found this mention [here](https://cloud.google.com/compute/docs/instances/specify-min-cpu-platform#how_selecting_a_minimum_cpu_platform_works). No behavior is specified for what happens when its not available, but apparently it errors out.

Its also reproducible with gcloud.

```
gcloud compute instances create example-instance2 --zone europe-west2-b --min-cpu-platform "Intel Haswell"
ERROR: (gcloud.compute.instances.create) Could not fetch resource:
 - Invalid value for field 'resource.minCpuPlatform': 'Intel Haswell'. CPU platform unavailable in requested zone. Valid options for this zone are: Intel Skylake, Intel Broadwell.
```

Sadness. `min_cpu_platform` looks rather pointless now.

**Comment by dghubble** [comment]
This would explain why setting `min_cpu_platform` did legitimately ensure Sandy/Ivy Bridge were skipped in other regions when this was verified, since Haswell was available.

**Comment by dghubble** [comment]
Fixed by 46196af5009fb2abef89a979248cb4bd55c6b941, which you may use as the ref in your module source.
